config/routes.rb

```diff
@@ -1,6 +1,18 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root "products#index"
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get "cart", to: "cart#show", as: 'cart'
+  delete "clean_cart", to: "cart#delete", as: "clean_cart"
+
+  resources :products, only: [:index, :show] do
+    member do
+      resource :cart do
+        [:add, :remove, :update_amount].each do |action|
+          post action, to: "cart#update", as: "#{action}_product_to"
```

Comment:
так навіть краще буде
```suggestion
          post action, to: "cart#update", as: "#{action}_product_in"
```